### PR TITLE
feat(rp-plate-solver): Phase 2 — crate scaffolding + AstapRunner + mock_astap

### PR DESCRIPTION
**Pull request by ivonnyssen**
## Summary

Phase 2 of [`docs/plans/rp-plate-solver.md`](docs/plans/rp-plate-solver.md) — establishes the `rp-plate-solver` workspace member with the trait surface, `.wcs` parser, subprocess supervision, and `mock_astap` test double. The HTTP server (`api.rs`) and `main.rs` are deferred to Phase 4.

## Library surface

- **`config`** — `Config`, `ConfigError`, `load_config` with startup validation (binary path executable, db dir is dir, timeout ordering, non-zero `max_concurrency`).
- **`error`** — `AppError` / `ErrorCode` / `ErrorResponse` mapping the five frozen error codes to HTTP status + snake_case JSON envelope.
- **`runner::AstapRunner`** — trait with `#[cfg_attr(test, mockall::automock)]` for upstream-of-runner unit tests.
- **`runner::astap::AstapCliRunner`** — real implementation. `build_command` is a pure function over `SolveRequest`; argv mapping is unit-tested. `ra_hint` converts decimal degrees → ASTAP hours; `dec_hint` converts to south-pole-distance. Windows: `CREATE_NEW_PROCESS_GROUP` at spawn so `CTRL_BREAK_EVENT` targets only the child.
- **`runner::wcs`** — minimal hand-rolled FITS-card parser (~100 lines, exhaustive error tests). The plan called for `fitsrs + wcs` but ASTAP's `.wcs` is header-only; until the Phase 2 spike validates `fitsrs` against a real ASTAP output, this stays as the implementation. Public surface (`read_wcs_sidecar`) doesn't change when the spike retires.
- **`supervision::spawn_with_deadline`** — races `child.wait()` against the deadline. On expiry: graceful signal → 2 s grace → force-kill (SIGTERM/SIGKILL on Unix; CTRL_BREAK_EVENT/TerminateProcess on Windows). Always `wait()`s child fully. `kill_on_drop(true)` is the safety-net layer.

## Test surface

- **26 unit tests** in-source: config validation, error mapping, `.wcs` parsing edge cases (missing CRVAL1/2/CDELT1, missing CROTA2 → defaults to 0, non-numeric → named error, extra keys ignored, ASTAP banner detection), argv mapping for every hint combination, supervision pure logic.
- **3 integration tests** in `tests/supervision_integration.rs` — Exited / TimedOutTerminated / TimedOutKilled arms against the real `mock_astap` child. Discovers the binary via `MOCK_ASTAP_BINARY` (Bazel) → `option_env!("CARGO_BIN_EXE_mock_astap")` (Cargo).
- **`mock_astap` `[[bin]]`** with six `MOCK_ASTAP_MODE` modes (`normal` / `exit_failure` / `hang` / `ignore_sigterm` / `malformed_wcs` / `no_wcs`) plus `MOCK_ASTAP_ARGV_OUT` side-channel. Mirrors `services/phd2-guider/src/bin/mock_phd2.rs`.

## Workspace

- `Cargo.toml`: registers `services/rp-plate-solver`, adds `wcs = "0.4"` (already transitive via `fitsrs` → `rp-fits`).
- `BUILD.bazel` for the new crate.
- `README.md` — operator install pointers per platform.

## Test plan

- [x] `cargo build -p rp-plate-solver --all-features --all-targets` clean
- [x] `cargo nextest run -p rp-plate-solver --all-features --all-targets` — 33 pass
- [x] `cargo clippy -p rp-plate-solver --all-targets -- -D warnings` clean
- [x] `cargo fmt --check` clean
- [x] `cargo rail run --profile commit -q` — 1115 tests pass across affected packages
- [ ] CI green
- [ ] Copilot review (request via UI)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: services/rp-plate-solver/src/supervision.rs [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        _ = tokio::time::sleep(deadline) => {
+            // Deadline. Send graceful signal, wait grace period, escalate.
+            send_graceful(pid);
+            match tokio::time::timeout(GRACE_PERIOD, child.wait()).await {
+                Ok(_status) => Ok(SpawnOutcome::TimedOutTerminated),
+                Err(_) => {
+                    // Force-kill. tokio's Child::kill sends SIGKILL on Unix
+                    // and TerminateProcess on Windows.
+                    let _ = child.start_kill();
+                    let _ = child.wait().await;
```
[the remark this answers is not in the document]
Comment (ivonnyssen) [reply services/rp-plate-solver/src/supervision.rs]: Confirmed deadlock — `wait()` blocks until exit, but a child writing >64 KiB to stderr fills the OS pipe buffer and blocks itself, never reaching exit. Fix: drain stderr concurrently in a spawned task before the deadline race. The task reads up to 4 KiB into a buffer then discards subsequent bytes (so the pipe never fills) and returns the captured tail when done. Awaited at the end of `spawn_with_deadline`.

File: services/rp-plate-solver/src/supervision.rs [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    };
+    let mut buf = Vec::with_capacity(4096);
+    let _ = s.read_to_end(&mut buf).await;
+    let text = String::from_utf8_lossy(&buf).to_string();
+    // Truncate to last 4096 bytes if very long.
+    if text.len() > 4096 {
+        text[text.len() - 4096..].to_string()
+    } else {
+        text
+    }
```
[the remark this answers is not in the document]
Comment (ivonnyssen) [reply services/rp-plate-solver/src/supervision.rs]: Right — `read_to_end` is unbounded. Restructured: the concurrent drain task above caps the *captured* buffer at 4 KiB inline (a `n.min(4096 - buf.len())` slice copy), then keeps reading into a discard buffer just to keep the pipe drained. Memory bounded at ~4 KiB regardless of stderr volume.

File: services/rp-plate-solver/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+[dependencies]
+tokio = { workspace = true }
+serde = { workspace = true }
+serde_json = { workspace = true }
+humantime = { workspace = true }
+humantime-serde = { workspace = true }
+axum = { workspace = true }
+tracing = { workspace = true }
+tracing-subscriber = { workspace = true }
```
[the remark this answers is not in the document]
Comment (ivonnyssen) [reply services/rp-plate-solver/Cargo.toml]: Removing `tracing-subscriber` for now — it'll come back in Phase 4 when `main.rs` initializes the subscriber. Same for `tracing` if there's nothing logging yet (keeping for now since `supervision::send_graceful` uses `tracing::debug!`).

File: services/rp-plate-solver/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+thiserror = { workspace = true }
+async-trait = { workspace = true }
+fitsrs = { workspace = true }
+wcs = { workspace = true }
+clap = { workspace = true }
```
[the remark this answers is not in the document]
Comment (ivonnyssen) [reply services/rp-plate-solver/Cargo.toml]: Removing `fitsrs`, `wcs`, and `clap`:
- `fitsrs`/`wcs` will return in Phase 2's spike-completion follow-up (or Phase 4) once we validate `fitsrs` against a real ASTAP `.wcs`.
- `clap` was anticipating Phase 4's `main.rs` CLI; defer until then.

The decision to *eventually* use `fitsrs+wcs` stays — it's an unused dep entry today, not a strategic change.

File: services/rp-plate-solver/src/supervision.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ivonnyssen) [reply services/rp-plate-solver/src/supervision.rs]: Confirmed UTF-8 boundary bug — `text.len() - 4096..` is a byte index that can split a multi-byte char (and `from_utf8_lossy` produces multi-byte replacement chars too). Fix is the same restructure as comment 1: keep the bounded buffer as `Vec<u8>` from the start, and call `from_utf8_lossy` once at the end on the already-bounded slice. No mid-string truncation, no boundary risk.

File: services/rp-plate-solver/src/supervision.rs [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+                Ok(0) => break,
+                Ok(n) => {
+                    if captured.len() < STDERR_TAIL_BYTES {
+                        let take = n.min(STDERR_TAIL_BYTES - captured.len());
+                        captured.extend_from_slice(&chunk[..take]);
+                    }
+                    // Past the limit: continue draining the pipe so the
+                    // child is never blocked on a full buffer, but
+                    // discard the bytes.
+                }
```
[the remark this answers is not in the document]
Comment (ivonnyssen) [reply services/rp-plate-solver/src/supervision.rs]: Right — `tail` claims trailing bytes but the implementation captured prefix bytes. Fix: actual ring-buffer-style tail capture. Read into a `Vec<u8>` that keeps only the last `STDERR_TAIL_BYTES` bytes; periodically `drain(..)` the head when the buffer exceeds 2× the limit (amortizes the shift cost). Memory stays bounded ≤ 2× limit, output is genuinely the tail of stderr (where ASTAP's error context lives). The wire-format field name `stderr_tail` and the `STDERR_TAIL_BYTES` const stay — they're now honest.

File: services/rp-plate-solver/tests/supervision_integration.rs [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+fn require_mock_astap() -> PathBuf {
+    mock_astap_path().unwrap_or_else(|| {
+        panic!(
+            "mock_astap binary not found.\n  \
+             Tried: MOCK_ASTAP_BINARY env var, then CARGO_BIN_EXE_mock_astap.\n  \
+             Under Cargo: run `cargo build --tests -p rp-plate-solver` first.\n  \
+             Under Bazel: set MOCK_ASTAP_BINARY in the test target's env."
+        )
+    })
```
[the remark this answers is not in the document]
Comment (ivonnyssen) [reply services/rp-plate-solver/tests/supervision_integration.rs:54]: Doc bug — the test fails (panics) when the binary isn't found, it doesn't skip. Updating the module doc comment to say "fails with a diagnostic" instead of "skip". A real skip path would need `#[ignore]` or runtime conditional `return`s, neither of which is the right behavior here: missing `mock_astap` *should* be a hard failure under both Cargo and Bazel, since the BUILD.bazel sets `MOCK_ASTAP_BINARY` and Cargo provides `CARGO_BIN_EXE_mock_astap`. If neither is set, something's wrong with the build.

File: services/rp-plate-solver/tests/supervision_integration.rs [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    let mut cmd = Command::new(&bin);
+    cmd.env("MOCK_ASTAP_MODE", mode);
+    // Mock binary doesn't actually need -f for the modes we exercise here
+    // (hang / ignore_sigterm), but pass one anyway so the argv shape is
+    // representative of the real call.
+    cmd.arg("-f").arg("/tmp/unused.fits");
+    #[cfg(windows)]
+    {
+        use std::os::windows::process::CommandExt;
+        cmd.creation_flags(0x00000200); // CREATE_NEW_PROCESS_GROUP
```
[the remark this answers is not in the document]
Comment (ivonnyssen) [reply services/rp-plate-solver/tests/supervision_integration.rs]: Confirmed magic-number drift. Adding `const CREATE_NEW_PROCESS_GROUP: u32 = 0x0000_0200;` at the top of both integration test files (matching `runner/astap.rs`'s style) so the constant is named consistently across the three call sites.

File: services/rp-plate-solver/src/runner/astap.rs [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+    #[test]
+    fn wcs_sidecar_path_replaces_fits_extension() {
+        assert_eq!(
+            wcs_sidecar_path(Path::new("/data/m31.fits")),
+            PathBuf::from("/data/m31.wcs")
+        );
+        assert_eq!(
+            wcs_sidecar_path(Path::new("/data/m31.fit")),
+            PathBuf::from("/data/m31.wcs")
+        );
```
[the remark this answers is not in the document]
Comment (ivonnyssen) [reply services/rp-plate-solver/src/runner/astap.rs:237]: Good catch — the docstring overclaimed. `Path::with_extension("wcs")` only replaces the *last* component, so `test.fits.fz` would become `test.fits.wcs` rather than `test.wcs`. Until we verify ASTAP's actual behavior on `.fits.fz` inputs (none of our current fixtures use that form), I'm dropping `.fz` from the docstring rather than claiming support we haven't verified. Adding a `.fz`-only test case (e.g., `test.fz` → `test.wcs`) since `with_extension` does handle that form correctly. The compound `.fits.fz` case becomes a documented non-support that retires when a real fixture surfaces.
